src/methods/construct.js

```diff
@@ -133,9 +134,9 @@ export function SpeciesConstructor(realm: Realm, O: ObjectValue, defaultConstruc
 }
 
 // ECMA 9.2.9
-export function MakeClassConstructor(realm: Realm, F: FunctionValue) {
+export function MakeClassConstructor(realm: Realm, F: ECMAScriptFunctionValue) {
   // 1. Assert: F is an ECMAScript function object.
-  invariant(F instanceof FunctionValue, "expected function value");
+  invariant(F instanceof ECMAScriptFunctionValue, "expected function value");
```

Comment:
Good catch, fixing the invariant. I see a few more instances of this kind of Assert in `construct.js` and `function.js`. Please change all of them, and push up the type signatures as appropriate.